### DISCUSSION TITLE
FaceResult

### DISCUSSION THREAD

**Discussion by pgg** [vladmandic/human#369] [Q&A]
hi vlad, For any given ```FaceResult``` why is ```size``` always: ```[0,0]``` and ```race``` and ```distance``` always ```undefined```?

**Comment by pgg** [comment]
is it because i have attention, iris and gear disabled?

**Reply by vladmandic** [reply]
yes, that's exactly it. distance is based on iris size - i've written about that before, its due to fact that iris size is relatively constant feature on all adult faces regardless of gender or race.
and race is part of gear model which is not enabled by default as its not as good as default one for age and its slower.

**Comment by pgg** [comment]
sorry, i should have checked before asking first, which i did just after asking. i enabled attention, iris and gear (had to specify gear model path as you don't have a default in config like you do for the others). now i get race and distance, but size is still [0,0]

**Comment by vladmandic** [comment]
i'll check.

btw, attention gives you more precise outlines around eyes and lips, but it comes at performance cost as well as compatibility issues - it uses operations that are not available in all backends and/or platforms.

**Reply by pgg** [reply]
ok, thanks for the info!

**Comment by pgg** [comment]
unrelated, but the following:
```
const result = await this.human.detect(this.human.webcam.element);
const context = result.canvas?.getContext('2d', { willReadFrequently: true });
const imageData = context?.getImageData(0, 0, result.canvas?.width ?? 0, result.canvas?.height ?? 0);
```
chrome complains with:

Canvas2D: Multiple readback operations using getImageData are faster with the willReadFrequently attribute set to true. See: https://html.spec.whatwg.org/multipage/canvas.html#concept-canvas-will-read-frequently

is there something special about the canvas in result?

**Comment by vladmandic** [comment]
ok, i've fixed face.size - it was simply not added to result depending on some configuration value combination.
regarding canvas `willReadFrequently`, that setting only works when context is first created. if you try to set that flag on a canvas that already has a context, you get the warning you see. and context is created first time deep inside the library. i've acually set it there as well now so you shouldn't do it from the user app.

new version will be published on github/npmjs later today.

**Reply by pgg** [reply]
ok, thanks, good to know

**Reply by pgg** [reply]
still getting canvas missing willReadFrequently with human 3.0.7

**Reply by pgg** [reply]
size works now, how does it relate to Box width (box[2]) and height (box[3])?

**Reply by vladmandic** [reply]
It's calculated from that.

**Reply by pgg** [reply]
whats the calculation? i am getting different values . . .
